src/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -89,6 +89,11 @@
     /** Set this factory's {@link ValidationStringency} to the provided one, then returns itself. */
     abstract public SamReaderFactory validationStringency(final ValidationStringency validationStringency);
 
+    /** Set whether readers created by this factory will use asynchronous IO.
+     * If this methods is not called, this flag will default to the value of {@link Defaults#USE_ASYNC_IO_FOR_SAMTOOLS}.
+     * Returns the factory itself. */
+    abstract public SamReaderFactory setUseAsyncIo(final boolean asynchronousIO);
```

Comment:
Does this work for all kinds of readers that could possibly be returned from this factory? Document.